### ISSUE TITLE
Build not triggered

### ISSUE THREAD

**Issue by pohmelie** [buildbot/buildbot#3563]
Config with multiple schedulers does not trigger build. If I remove one scheduler, then build triggered. But for two schedulers no build triggered.
``` python
from buildbot.plugins import worker, changes, schedulers, util, steps


REPO = "git@git.int.kronshtadt.ru:ktrs-sdc/test-python-buildbot.git"
c = BuildmasterConfig = {}
c["workers"] = [worker.LocalWorker("worker", max_builds=1)]
c["protocols"] = {"pb": {"port": 9989}}
c["change_source"] = [changes.GitPoller(repourl=REPO, branches=["master"], pollInterval=10)]
c["schedulers"] = [
    schedulers.AnyBranchScheduler(name="s1", builderNames=["builder"], properties=dict(foo="bar")),
    schedulers.AnyBranchScheduler(name="s2", builderNames=["builder"], properties=dict(foo="foo")),
]
factory = util.BuildFactory([steps.ShellCommand(name="test", command=["echo", "test"])])
c["builders"] = [util.BuilderConfig(name="builder", workernames=["worker"], factory=factory)]
c["services"] = []
c["title"] = "Test buildbot"
c["buildbotURL"] = "http://localhost:8010/"
c["www"] = dict(port=8010, plugins=dict(waterfall_view={}, console_view={}, grid_view={}))
c["db"] = dict(db_url="sqlite:///state.sqlite")
```
```
2017-08-29 14:59:31+0300 [-] gitpoller: processing changes from "git@git.int.kronshtadt.ru:ktrs-sdc/test-python-buildbot.git"
2017-08-29 14:59:41+0300 [-] gitpoller: processing changes from "git@git.int.kronshtadt.ru:ktrs-sdc/test-python-buildbot.git"
2017-08-29 14:59:41+0300 [-] gitpoller: processing 1 changes: ['20e32aca133787a9abfa91896d27bfd30249c7d7'] from "git@git.int.kronshtadt.ru:ktrs-sdc/test-python-buildbot.git" branch "master"
2017-08-29 14:59:41+0300 [-] added change with revision 20e32aca133787a9abfa91896d27bfd30249c7d7 to database
2017-08-29 14:59:42+0300 [-] added buildset 1 to database
2017-08-29 14:59:42+0300 [-] added buildset 2 to database
2017-08-29 14:59:51+0300 [-] gitpoller: processing changes from "git@git.int.kronshtadt.ru:ktrs-sdc/test-python-buildbot.git"
```

**Comment by seankelly** [comment]
It does list two buildsets being added. When you delete one of the schedulers, is only one buildset added then?

**Comment by pohmelie** [comment]
@seankelly, yes.

**Comment by p12tic** [comment]
This bug is very likely to be a duplicate of #4642 which has been fixed in #6152. I'm thus closing the issue.

If the problem is still present after Buildbot v3.3 with #6152 is released, please open another issue as the cause is likely to be different.

Thanks a lot for reporting the bug.
